[longnguyentan] iP

longnguyentan opened the pull request:
# Dialogix

> "The only limit to our realization of tomorrow will be our doubts of today." – Franklin D. Roosevelt ([source](https://www.brainyquote.com/quotes/francois_de_la_rochefouca_164188))

Dialogix liberates your mind from the burden of remembering tasks. It offers:

- Text-based interface
- Intuitive learning curve
- Lightning-fast usability

To get started:

1. [Grab it here](https://github.com/longnguyentan/ip/releases/tag/A-Jar).
2. Double-click the installer.
3. Input your to-dos.
4. Let it handle your tasks with a 😄!

And the best part? It's completely **FREE**!

**Key Features:**

- [x] Task management
- [x] Upcoming deadlines
- [ ] Reminder functionality (coming soon)

If you're a Java enthusiast, you can also utilize it for Java practice. Check out the `main` method:

```java
public class EntryPoint {
    public static void main(String[] args) {
        App.launch(DialogixApp.class, args);
    }
}

longnguyentan committed:
Level 0 Update
Rename, Greet, Exit

longnguyentan committed:
Level 0 Update
Rename, Greet, Exit

longnguyentan committed:
Level 1. Echo

longnguyentan committed:
Level 2. Add, List

longnguyentan committed:
Level 3. Mark as Done

longnguyentan committed:
Level 3. Mark as Done

longnguyentan committed:
Level 4. ToDos, Events, Deadlines

longnguyentan committed:
Level 5. Handle Errors

longnguyentan committed:
Level 5. Handle Errors

longnguyentan committed:
Level 6. Delete

longnguyentan committed:
A-TextUiTesting

longnguyentan committed:
Use Java enums

longnguyentan committed:
Update Save feature

longnguyentan committed:
Update Dates and Times

longnguyentan committed:
Implement Ui, Storage, Parser, TaskList

longnguyentan committed:
Add JUnit Tests

longnguyentan committed:
Update Find

longnguyentan committed:
Update commands and add checkstyle

longnguyentan committed:
Update GUI

longnguyentan committed:
Update GUI

saltedfishxx commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,184 @@
+        } else {
+            // throw new DialogixException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+        return null;
```
You can consider simplfying the code by using [enums](https://www.w3schools.com/java/java_enums.asp) and [switch cases](https://www.w3schools.com/java/java_switch.asp). This not only reduces the number of conditionals and braces for better readability, but also introduces lesser repetitive methods, such as `isDeleteCommand()`, `isFindCommand()` to maintain!

saltedfishxx commented on src/test/java/UiTest.java:
```diff
@@ -0,0 +1,34 @@
+        // Restore System.out
+        System.setOut(System.out);
+
+        String expectedOutput = "____________________________________________________________\n" +
+            "Hello! I'm Dialogix\n" +
+            "What can I do for you?\n" +
+            "____________________________________________________________\n";
```
Can be a final String constant!

saltedfishxx reviewed (commented):
Overall good code quality with good OOP usage and methods. Keep up the good work!

tayian commented:
unique choice of name for your project (bonus points for creativity!) thumbs up for your implementation of OOP. hope this tool ends up being useful for you as well!

tayian commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,184 @@
+ * Command
+ */
+public class Parser {
+    public static Command parse(String fullCommand) throws DialogixException {
```
building on saltedfishxx's point, you might be able to shorten the lines of if-else statements that follow this by working through the logic you use.

and for this entire class, might I suggest using JavaDoc (or just // comments to provide more code clarity?)

tayian commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+    public List<Task> findTasks(String keyword) {
+        return tasks.stream()
+            .filter(task -> task.getDescription().contains(keyword))
+            .collect(Collectors.toList());
```
interesting implementation of findTasks

longnguyentan committed:
Update Commands

longnguyentan committed:
Update Launcher

longnguyentan committed:
Improve code quality in all packages

The Dialogix project has some code quality issues, including inconsistent formatting, long methods, and missing JavaDoc comments.

These code quality issues make the project hard to maintain and can lead to bugs and readability problems.

- Refactored the code to break down long methods into smaller, more manageable functions.
- Added JavaDoc comments to enhance code documentation and readability.
- Improved formatting to adhere to coding standards.

These changes are made to enhance code maintainability, readability, and overall quality. The refactoring and documentation improvements will make the codebase more understandable and maintainable for future development.

longnguyentan committed:
Improve code quality in all packages

The Dialogix project has some code quality issues, including inconsistent formatting, long methods, and missing JavaDoc comments.

These code quality issues make the project hard to maintain and can lead to bugs and readability problems.

- Refactored the code to break down long methods into smaller, more manageable functions.
- Added JavaDoc comments to enhance code documentation and readability.
- Improved formatting to adhere to coding standards.

These changes are made to enhance code maintainability, readability, and overall quality. The refactoring and documentation improvements will make the codebase more understandable and maintainable for future development.

longnguyentan committed:
Write tests with Assertions

Write test cases for the classes in Command package to use Java's assert feature for documenting and validating important assumptions within the code. This improves code readability and helps ensure that key conditions are met during testing.

This change involves adding assert statements to existing test cases for the Todo class and introducing new test cases that leverage the assert feature to validate different scenarios. The use of assertions enhances the clarity of the tests and makes it easier to identify issues when they occur.

These changes adhere to the best practice of using assertions to document and verify critical assumptions within the codebase.

longnguyentan committed:
Update PriorityCommand

longnguyentan committed:
Merge pull request #2 from longnguyentan/branch-O-Optional

Update PriorityCommand

longnguyentan committed:
Update User Guide

longnguyentan committed:
Update Delete function

longnguyentan committed:
Update Save function

longnguyentan committed:
Update README.md

longnguyentan committed:
Update Find command

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Trigger rebuild

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md

longnguyentan committed:
Update README.md
